src/compiler/checker.ts

```diff
@@ -43532,7 +43532,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     // entire parameter does not have type annotation, suggest adding an annotation
                     addRelatedInfo(
                         diagnostic,
-                        createFileDiagnostic(getSourceFileOfNode(wrappingDeclaration), wrappingDeclaration.end, 1, Diagnostics.We_can_only_write_a_type_for_0_by_adding_a_type_for_the_entire_parameter_here, declarationNameToString(node.propertyName)),
+                        createFileDiagnostic(getSourceFileOfNode(wrappingDeclaration), wrappingDeclaration.end, 0, Diagnostics.We_can_only_write_a_type_for_0_by_adding_a_type_for_the_entire_parameter_here, declarationNameToString(node.propertyName)),

```

Comment:
There are other file diagnostics with `length: 0` so this isn't something new. I'm not sure what exact impact this might have on IDEs though 🤷‍♂️ If necessary I could detect a situation when this would exceed the source file's length and adjust the start position and the length but that would produce a worse location for this (edge) case. So it feels like this is actually better right now.

Alternatively, this could be node diagnostic with an adjusted message or smth? But then the start location of this diagnostic would still be way earlier than the location at which this error suggests that a type annotation should be inserted.

There are also other places that might be affected by similar bugs and they could be investigated. Especially some calls to `grammarErrorAtPos` might suffer from this. That's a separate issue though - I don't have any test cases for them at hand

Reply:
Length 1 and length 0 in my experience show up the same in editors; as a single character squiggle. So, it should be okay.

This makes me wonder what other length 1s we have that are problematic.